infra/src/main/java/com/redhat/service/smartevents/infra/models/gateways/Gateway.java

```diff
@@ -1,25 +1,20 @@
-package com.redhat.service.smartevents.infra.models.actions;
+package com.redhat.service.smartevents.infra.models.gateways;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
-import javax.validation.constraints.NotEmpty;
-import javax.validation.constraints.NotNull;
-
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class BaseAction {
+public class Gateway {
```

Comment:
What about a javadoc/readme to explain the Gateway abstraction/metaphor?

Reply:
I added a javadoc with the same explanation I used in the description of this PR :+1: